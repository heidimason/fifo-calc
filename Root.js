import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { combineReducers } from 'redux'

import PurchaseHistoryReducer from './src/reducers/PurchaseHistory'
import SaleHistoryReducer from './src/reducers/SaleHistory'

const rootReducer = combineReducers({
		purchaseHistory: PurchaseHistoryReducer,
		saleHistory: SaleHistoryReducer
	}),

	store = createStore(
		rootReducer
	)

export default props => {
	return (
		<Provider store={store}>
			{props.children}
		</Provider>
	)
}
