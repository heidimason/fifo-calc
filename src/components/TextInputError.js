import React from 'react'
import { Text } from 'react-native'
import { orange } from '../utils/styles/colors'
import { fonts } from '../utils/styles/fonts'
import styled from 'styled-components/native'

const TextInputError = ({text}) => {
	return (
		<ErrorText
			style={fonts.h3}>{text}
		</ErrorText>
	)
}

const ErrorText = styled.Text`
    color: ${orange}
    text-align: center
    margin-bottom: 15
`

export default TextInputError
