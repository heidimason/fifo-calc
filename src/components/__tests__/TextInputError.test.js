import React from 'react'
import renderer from 'react-test-renderer'

import Root from '../../../Root'
import TextInputError from '../TextInputError'

describe('<TextInputError />', () => {
	const tree = renderer.create(
		<Root>
			<TextInputError />
		</Root>
	).toJSON()

	it('renders correctly', () => {
		expect(tree).toMatchSnapshot()
	})
})
