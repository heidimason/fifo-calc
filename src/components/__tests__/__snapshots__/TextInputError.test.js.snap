// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInputError /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffa500",
        "marginBottom": 15,
        "textAlign": "center",
      },
      Object {
        "fontSize": 27,
      },
    ]
  }
/>
`;
